src/Swagger/Serializer/DocumentationNormalizer.php

```diff
@@ -56,185 +57,265 @@ public function __construct(ResourceMetadataFactoryInterface $resourceMetadataFa
 
-            $operations = $resourceMetadata->getCollectionOperations() ?? [];
-            foreach ($operations as $operationName => $collectionOperation) {
+            foreach ($resourceMetadata->getCollectionOperations() ?? [] as $operationName => $operation) {
+                $path = $this->getPath($resourceShortName, $operation, true);
                 $method = $this->operationMethodResolver->getCollectionOperationMethod($resourceClass, $operationName);
-                $path = $this->getPath($shortName, $collectionOperation, true);
 
-                $swaggerOperation = $this->getSwaggerOperation($resourceClass, $resourceMetadata, $collectionOperation, true, $method, $object->getMimeTypes());
-                $itemOperationsDocs[$path] = array_merge($itemOperationsDocs[$path] ?? [], $swaggerOperation);
+                $paths[$path][strtolower($method)] = $this->getPathOperation($operationName, $operation, $method, true, $resourceMetadata, $object->getMimeTypes());
             }
 
-            $operations = $resourceMetadata->getItemOperations() ?? [];
-            foreach ($operations as $operationName => $itemOperation) {
+            foreach ($resourceMetadata->getItemOperations() ?? [] as $operationName => $operation) {
+                $path = $this->getPath($resourceShortName, $operation, false);
                 $method = $this->operationMethodResolver->getItemOperationMethod($resourceClass, $operationName);
-                $path = $this->getPath($shortName, $itemOperation, false);
 
-                $swaggerOperation = $this->getSwaggerOperation($resourceClass, $resourceMetadata, $itemOperation, false, $method, $object->getMimeTypes());
-                $itemOperationsDocs[$path] = array_merge($itemOperationsDocs[$path] ?? [], $swaggerOperation);
+                $paths[$path][strtolower($method)] = $this->getPathOperation($operationName, $operation, $method, false, $resourceMetadata, $object->getMimeTypes());
             }
         }
 
-        return $this->computeDoc($object, $definitions, $itemOperationsDocs);
+        return $this->computeDoc($object, $definitions, $paths);
     }
 
+    /**
+     * Gets the path for an operation.
+     *
+     * If the path ends with the optional _format parameter, it is removed
+     * as optional path parameters are not yet supported.
+     *
+     * @see https://github.com/OAI/OpenAPI-Specification/issues/93
+     *
+     * @param string $resourceShortName
+     * @param array  $operation
+     * @param bool   $collection
+     *
+     * @return string
+     */
     private function getPath(string $resourceShortName, array $operation, bool $collection) : string
     {
         $path = $this->operationPathResolver->resolveOperationPath($resourceShortName, $operation, $collection);
-        if (substr($path, -10) === '.{_format}') {
+        if ('.{_format}' === substr($path, -10)) {
```

Comment:
Yes, there are many other issues I've identified. But this PR is just for refactoring / better code quality.